File: tests/test-node-app/src/functions/blobProcessor.js
const { app } = require('@azure/functions');

app.storageBlob('blobProcessor', {
    path: 'test-container/{name}',
    connection: 'AzureWebJobsStorage',
    handler: async (blob, context) => {
        context.log(`Blob trigger processed: ${context.triggerMetadata.name}, Size: ${blob.length} bytes`);
    }
});

File: tests/test-dotnet-isolated-app/BlobProcessor.cs
using Microsoft.Azure.Functions.Worker;
using Microsoft.Extensions.Logging;

namespace TestDotnetIsolated;

public class BlobProcessor
{
    private readonly ILogger<BlobProcessor> _logger;

    public BlobProcessor(ILogger<BlobProcessor> logger)
    {
        _logger = logger;
    }

    [Function("blobProcessor")]
    public void Run(
        [BlobTrigger("test-container/{name}", Connection = "AzureWebJobsStorage")] byte[] blob,
        string name)
    {
        _logger.LogInformation($"Blob trigger processed: {name}, Size: {blob.Length} bytes");
    }
}
